test/transaction.test.js

```diff
@@ -0,0 +1,175 @@
+  function createPostInTx(post, timeout) {
+    return function(done) {
+      // Transaction.begin(db.connector, Transaction.READ_COMMITTED,
+      Post.beginTransaction({
+          isolationLevel: Transaction.READ_COMMITTED,
+          timeout: timeout
+        },
+        function(err, tx) {
+          if (err) return done(err);
+          expect(typeof tx.id).to.eql('string');
+          hooks = [];
+          tx.observe('before commit', function(context, next) {
+            hooks.push('before commit');
+            next();
+          });
+          tx.observe('after commit', function(context, next) {
+            hooks.push('after commit');
+            next();
+          });
+          tx.observe('before rollback', function(context, next) {
+            hooks.push('before rollback');
+            next();
+          });
+          tx.observe('after rollback', function(context, next) {
+            hooks.push('after rollback');
+            next();
+          });
+          currentTx = tx;
+          Post.create(post, {transaction: tx},
+            function(err, p) {
+              if (err) {
+                done(err);
+              } else {
+                done();
+              }
+            });
+        });
+    };
+  }
+
+  // Return an async function to find matching posts and assert number of
+  // records to equal to the count
+  function expectToFindPosts(where, count, inTx) {
+    return function(done) {
+      var options = {};
+      if (inTx) {
+        options.transaction = currentTx;
+      }
+      Post.find({where: where}, options,
+        function(err, posts) {
+          if (err) return done(err);
+          expect(posts.length).to.be.eql(count);
+          done();
+        });
+    };
+  }
+
+  describe('commit', function() {
+
+    var post = {title: 't1', content: 'c1'};
+    before(createPostInTx(post));
+
+    it('should not see the uncommitted insert', expectToFindPosts(post, 0));
+
+    it('should see the uncommitted insert from the same transaction',
+      expectToFindPosts(post, 1, true));
+
+    it('should commit a transaction', function(done) {
+      currentTx.commit(function(err) {
+        expect(hooks).to.eql(['before commit', 'after commit']);
+        done(err);
+      });
+    });
+
+    it('should see the committed insert', expectToFindPosts(post, 1));
+
+    it('should report error if the transaction is not active', function(done) {
+      currentTx.commit(function(err) {
+        expect(err).to.be.instanceof(Error);
+        done();
+      });
+    });
+  });
+
+  describe('rollback', function() {
+
+    before(function() {
+      // Reset the collection
+      db.connector.collection.data = [];
+    });
+
+    var post = {title: 't2', content: 'c2'};
+    before(createPostInTx(post));
+
+    it('should not see the uncommitted insert', expectToFindPosts(post, 0));
+
+    it('should see the uncommitted insert from the same transaction',
+      expectToFindPosts(post, 1, true));
```

Comment:
Would be nice if the `expectToFindPosts` took an options object so the arguments made sense... not sure what `true` means here. Just for future notice... no need to change for this pr.